ruby/lib/transformer/syskit/frame_propagation.rb

```diff
@@ -224,38 +224,9 @@ def initial_information(task)
             end
         end
 
-        class PortAssociationMismatch < RuntimeError
-            # The problematic endpoint, as a [task, port_name] pair
-            attr_reader :endpoint
-            # The other side of the problematic connection(s)
-            attr_reader :connections
-            # The association type expected by +endpoint+. Can either be 'frame'
-            # for an association between a port and a frame, and 'transform' for
-            # an association between a port and a transformation.
-            attr_reader :association_type
-
-            def initialize(task, port_name, type)
-                @endpoint = [task, port_name]
-                @association_type = type
-
-                @connections = []
-                task.each_concrete_input_connection(port_name) do |source_task, source_port_name, _|
-                    @connections << [source_task, source_port_name]
-                end
-                task.each_concrete_output_connection(port_name) do |_, sink_port_name, sink_task, _|
-                    @connections << [sink_task, sink_port_name]
-                end
-            end
-
-            def pretty_print(pp)
-                pp.text "#{endpoint[0]}.#{endpoint[1]} was expecting an association with a #{association_type}, but one or more connections mismatch"
-                pp.nest(2) do
-                    pp.breakable
-                    pp.seplist(connections) do |conn|
-                        pp.text "#{conn[0]}.#{conn[1]}"
-                    end
-                end
-            end
+        # Check if there is a port association mismatch. This method exists mostly to be mocked.
+        def port_association_mismatch?(info)
+            !info.respond_to? :selected_frame
         end
```

Comment:
This is not a generic test for "port association mismatch". It is only valid in the context of the `each_annotated_port` loop.

The test below is checking that a port that is associated with a frame is not connected to a port that is associated with a transform. I don't think you really need mocking for this, do you really ?

Reply:
Nop, I had trouble recreating the error but managed to in the end